### DISCUSSION TITLE
Type of mutation input in 1.13.0

### DISCUSSION THREAD

**Discussion by pic** [rmosolgo/graphql-ruby#3799] [Q&A] [answered]
Hi,
upgrading from 1.12.21 to 1.13.x we have noticed the following change on mutations:

given the mutation

```
class Mutations::MyNamespace::MyMutation
end
```

it used to create a type for the input named: `MyNamespaceMyMutation`
After the upgrade, it's just `MyMutation`.

Is there a why to restore the old behavior or is it possible to explicitly define the name of the mutation input class?

Thanks

**Comment by rmosolgo** [comment, marked as answer]
Hi! Sorry to hear that it stopped working. 

There _were_ some changes to how mutations generate their input types, for example: https://github.com/rmosolgo/graphql-ruby/compare/v1.12.21...v1.13.0#diff-74f77a993e7facb851ef0be965716bac540dfe856aae65dff2f0c755ae26f1f0R86


I would _expect_ that `def self.default_graphql_name` in the mutation class be in charge of this, based on this code:

https://github.com/rmosolgo/graphql-ruby/blob/8bf636412c2598801c5dddc8b411c8f27d4c48cc/lib/graphql/schema/relay_classic_mutation.rb#L144-L147

So, I'd expect something like this to work:


```ruby 
class Mutations::BaseMutation < GraphQL::Schema::RelayClassicMutation 
  def self.default_graphql_name 
    # remove leading `Mutation` and any `::` from class name 
    @default_graphql_name ||= name.sub(/^Mutation/, "").gsub("::", "") 
  end
end 
```

Then, the code above would add `"Input"` to the end of the string returned there. 

Does that work for you? Or do you have any similar code in your application that _used_ to work, but doesn't work anymore? (If so, could you please share it? Maybe I've broken something 🙈 !)

**Reply by pic** [reply]
Thank you so much. Defining `self.default_graphql_name` solved the issue for us. Only it's `/^Mutations/` (with trailing s)
